### ISSUE TITLE
iPhone HEIC/HEIF image format read support

### ISSUE THREAD

**Issue by nincode** [darktable-org/darktable#4989]
**Is your feature request related to a problem? Please describe.**
Problem: iPhones these days generate heic files. I would like to be able to use darktable to organize heics.

**Describe the solution you'd like**
.heic read support in darktable, similar to other formats. Don't need to edit, just to catalog, organize, have xmp sidecars, etc.

**Alternatives**
Converting .heics to .jpg is possible, but you lose quality by recompressing, increase file size (an issue for large libraries), and (I believe) lose dynamic range,

**Additional context**
https://github.com/strukturag/libheif

**Comment by rfeg** [comment]
heic editing would be nice also (read is 1st priority, of course)

**Comment by 1div0** [comment]
https://redmine.darktable.org/issues/12244

**Comment by w-oertl** [comment]
This image format is in use not only by iOS/iPadOS (iPhone/iPad), but also Android phones by now. Reading is supported by libheif1 (available in Debian, see https://github.com/strukturag/libheif). It would be quite useful to have DT reading this file format.

**Comment by chhil** [comment]
Any timetable on this?

**Comment by johnny-bit** [comment]
this is open source project, there's hardly any timeline here.

First - we need isobmff support in exiv2 and as I'm aware @1div0 works with Robin on getting that into exiv2 then it's question of reading via external lib such as libheif and/or libavif

**Comment by 1div0** [comment]
Yes, at the moment I am reworking the metadata reading without any external libraries. No ETA.

**Comment by johnny-bit** [comment]
I wanted to work on that too, but since you took the responsibility... You plan on supporting full set of ISOBMFF files (so atm HEIF/HEIC/AVIF/CR3)? Would you want some help with testing etc?

**Comment by 1div0** [comment]
So far I am focused on the HEIF only.

Please have a look at https://github.com/1div0/exiv2/commit/0ad0cd68f33968ac9b2f38f9650aa5922b178951

Far from perfect, just restarted after long time.

**Comment by 1div0** [comment]
peter.kovar@Pascal /1TB/usr/src/github.com/1div0/exiv2/Linux/x86-64
€ bin/exiv2 ~/Obrázky/HEIF/IMGP0004.heif 
File name       : /home/peter.kovar/Obrázky/HEIF/IMGP0004.heif
File size       : 2652963 Bytes
MIME type       : image/heif
Image size      : 0 x 0
/home/peter.kovar/Obrázky/HEIF/IMGP0004.heif: No Exif data found in the file

**Comment by 1div0** [comment]
I am wondering if anyone can find .heix files with 10 bpc or higher bit depth. Just want to make sure the detection works reliably and quickly.

**Comment by johnny-bit** [comment]
If you meant heic files I only saw things in https://github.com/tigranbs/test-heic-images https://nokiatech.github.io/heif/examples.html https://www.heicconvert.com/examples nothing else.

**Comment by 1div0** [comment]
@johnny-bit these are old and not compatible with recent iOS reader/writer anymore.

**Comment by chhil** [comment]
@1div0 
Would a file with these properties work?

![image](https://user-images.githubusercontent.com/1119439/100231225-b8d63c80-2f4c-11eb-8b53-8aef9a8875f6.png)

**Comment by chhil** [comment]
The file referenced above
https://www.dropbox.com/s/6nh03oj1dinxnqg/Photo%2021-11-20%2C%205%2027%2004%20PM.heic?dl=0

**Comment by 1div0** [comment]
Yes, indeed.

```
€ imlib2_view ~/Obrázky/HEIF/Photo\ 21-11-20\,\ 5\ 27\ 04\ PM.heic 
 * Resolution     : 3024x4032
 * Bit Depth      : 24

**Comment by 1div0** [comment]
Bit depth equal to 32 is misleading in W10. We are talking about 8 bpc here, that means ordinary 24 bpp.

**Comment by rfeg** [comment]
Canon cameras have 10bit depth, better to allow higher bit depths than 8 (ideally up to 16) to be future proof.
reference:  https://petapixel.com/2019/10/31/canon-has-moved-on-to-heif-but-wont-ditch-jpeg-completely-yet/

**Comment by Fiedzia** [comment]
Sony A7S III and α1 generate heif too, 10bit.

**Comment by cytrinox** [comment]
I'm working on a PR to support HEIF via **libheif**. I've found two sample files online and they are working (from Apple iPhone and Canon R5). I need more test files, so @Fiedzia can you share samples under CC0 license? Please make sure (if your camera support it) to provide different files for different color profiles (like sRGB, PQ2020, AdobeRGB). The file name should include what it is (sony_a7s_3_heif_10bit_PQ2020.heif for example)
